pkg/ingester/ingester.go

```diff
@@ -777,12 +776,12 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 		if errors.Is(err, errMaxTenantsReached) {
 			return nil, err
 		}
-		return nil, wrapWithUser(err, userID)
+		return nil, annotateWithUser(err, userID)
 	}
 
 	lockState, err := db.acquireAppendLock(req.MinTimestamp())
 	if err != nil {
-		return &mimirpb.WriteResponse{}, httpgrpc.Errorf(http.StatusServiceUnavailable, wrapWithUser(err, userID).Error())
+		return &mimirpb.WriteResponse{}, newValidationError(annotateWithUser(err, userID), http.StatusServiceUnavailable)
```

Comment:
I wouldn't call this "validation" error. "Validation" errors are result of validation against limits.

Reply:
I have introduced another error type, `errorWithStatus`, that will be used for wrapping errors returned by ingester. `validationError` will keep its original purpose, i.e., wrap errors obtained from validations against limits.